testing/common_testing_util.py

```diff
@@ -0,0 +1,110 @@
+from pathlib import Path
+import pytest
+from ivadomed.scripts import download_data as ivadomed_download_data
+import os
+import shutil
+import sys
+
+# Cannot use __ivadomed_dir__ here as that would refer to the IvadoMed installed in the Conda/Venv
+# Absolute path to the temporary folder, which should sit at the root of the repo.
+# We could also replace this hard coded levels reference with a simple loop to search for .git/.github sitting at the
+# root of the repo. To be discussed.
+FOLDER_LEVELS_FROM_ROOT = 1
+
+path_repo_root: Path = Path(__file__).parents[FOLDER_LEVELS_FROM_ROOT].absolute()
+
+path_temp: str = str(path_repo_root / "tmp")
+
+path_unit_tests: str = str(path_repo_root / "testing" / "unit_tests")
+path_data_testing_source: str = str(path_repo_root / "data_testing")
+path_data_testing_tmp: str = str(path_repo_root / "tmp" / Path(path_data_testing_source).name)
+
+path_functional_tests: str = str(path_repo_root / "testing" / "functional_tests")
+path_data_functional_source: str = str(path_repo_root / "data_functional_testing")
+path_data_functional_tmp: str = str(path_repo_root / "tmp" / Path(path_data_functional_source).name)
+
+
+def download_dataset(dataset: str = 'data_testing', verbose=True):
+    """Download testing data from internet.
+
+    Args:
+        verbose (bool): whether or not to print
+    """
+
+    path_dataset: Path = path_repo_root / dataset
+
+    # Early abort if testing data already exist.
+    if path_dataset.exists():
+        printv(f'\nTesting data files appear to already exist at {path_dataset}, aborting ddownload', verbose)
+        return
+
+    printv(f'\nDownloading testing data... to {dataset}', verbose)
+
+    # Call the ivadomed download CLI
+    ivadomed_download_data.main([
+        '-d', dataset,
+        '-o', str(path_dataset)
+    ])
+
+
+def remove_dataset(dataset: str = 'data_testing', verbose=True):
+    """Recursively remove the data_testing folder.
+
+    Args:
+        verbose (bool): whether or not to print
+    """
+    if os.path.exists(path_temp):
+        path_dataset = os.path.join(path_temp, dataset)
+    else:
+        path_dataset = os.path.join(path_temp, dataset)
+
```

Comment:
Seems like the behaviour here is the same whether the path exists or not. Perhaps this was an unintentional bug or an oversight?

Reply:
Good call. That was a bad refactoring on my part. I have fixed it to ensure its consistency. Curiously this function is rarely used anywhere (except in unit test). I wonder why. Might be part of old setup/teardown process that I need to further clean up in the PR in the future.  As for type hinting, see discussion proposal [here](https://docs.google.com/document/d/1wgKjqU1lGQanjXZxYzjjwp8vdA_QiJgQ88B5efjzT1w/edit?usp=sharing)